Sources/ContentfulPersistence/SynchronizationManager.swift

```diff
@@ -123,6 +128,68 @@ public class SynchronizationManager: PersistenceIntegration {
+    ///     This must not be `.all`.
+    ///   - limit: An optional limit for the number of records to synchronize in one batch. Defaults to `nil`, indicating no limit.
+    ///   - dbVersion: The database version targeted for synchronization. Defaults to the application's default database version.
+    ///   - onInitialCompletion: A closure called upon completion of the initial synchronization phase, with a `Result` containing either the `SyncSpace` or an error.
+    ///   - onFinalCompletion: A closure called upon completion of the final synchronization phase, with a `Result` containing either the `SyncSpace` or an error.
+    /// - Throws: An assertion failure if `initialLocalizationScheme` is `.all`.
+    ///
+    /// - The synchronization process involves two phases:
+    ///     1. Synchronizing data with the provided localization scheme.
+    ///     2. Resetting the synchronization token and synchronizing with `.all` localizations.
+    ///     The `onInitialCompletion` closure is invoked after the first phase, and the final result is passed to the `onFinalCompletion` closure.
+    public func sync<SSP>(
+        syncSpacePersistable: SSP.Type,
+        initialLocalizationScheme: LocalizationScheme,
+        limit: Int? = nil,
+        dbVersion: Int = DBVersions.default.rawValue,
+        onInitialCompletion: @escaping ResultsHandler<SyncSpace>,
+        onFinalCompletion: @escaping ResultsHandler<SyncSpace>
+    ) throws where SSP: SyncSpacePersistable  {
+        // Ensure the initial localization scheme is valid
+        if case .all = initialLocalizationScheme {
+            throw Error.invalidLocalizationScheme
+        }
+
+        // Closure to handle intermediate sync completion
+        let handlePartialCompletion: ResultsHandler<SyncSpace> = { [weak self] result in
+            // Invoke the initial completion handler with the result
+            onInitialCompletion(result)
+
+            // Ensure `self` is still available
+            guard let self = self else {
+                onFinalCompletion(.failure(Error.deallocatedClient))
+                return
+            }
+
+            // Reset the synchronization token
+            do {
+                let syncInfo: SSP = try persistentStore.fetchOne(
+                    type: syncSpacePersistable,
+                    predicate: NSPredicate(value: true)
+                )
+                syncInfo.syncToken = nil
+                save()
+            } catch {
+                onFinalCompletion(.failure(error))
+                return
+            }
+
+            // Proceed to sync with `.all` localizations
+            self.localizationScheme = .all
+            self.sync(limit: limit, dbVersion: dbVersion, then: onFinalCompletion)
+        }
+
+        // Begin the initial synchronization
+        self.localizationScheme = initialLocalizationScheme
+        sync(limit: limit, dbVersion: dbVersion, then: handlePartialCompletion)
```

Comment:
Would it be more efficient to go through all locales one by one and sync them rather than doing "sync all" which would resync the same locale which was synced in the first place?

Reply:
I was wrapping my head around this, because I agree and don't particularly like the fact that we are syncing the same locale twice.

On he other hand, we need a way to store a valid sync token that can be used for subsequent sync operations.

To decouple the sync token from the "all", we would need a "per-locale" syncToken. I think that makes the whole system even more complex.

What do you think?